sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,201 @@
+from sympy.combinatorics.free_groups import free_group, FreeGroupElement
+from sympy.combinatorics.fp_groups import FpGroup, FpSubgroup, subgroup_quotient
+from sympy import sieve
+
+class PcGroup(FpGroup):
+    '''
+    Class implementing Polycyclic groups.
+    To-do : Docstring completion.
+            Add error and exception handling
+    The generators will be converted to equivalent forms on
+    another free group.
+    The implementation for this methods will be similar to
+    the coin change problem in DP.
+    '''
+    def __init__(self, pc_series, power_exponents=None):
+        self.pc_series = pc_series
+        self.fp_group = self.pc_series[0]
+        self.pc_gens = self.compute_pcgs()
+        self.rel_orders = self.relative_orders()
+        if not power_exponents:
+            self.power_exponents = self.relative_orders
+        self.power_relations = {}
+        self.conjugate_relations = {}
+        self.relations = self.power_relations + self.conjugate_relations
+        # All the elements of the `pc_gens` will be replaces
+        # by the elements of another `FreeGroup`.
+        self.F, self.x, self.pc_gens_dict = self._pc_gens_dict()
+
+    def _pc_gens_dict(self):
+        '''
+        Map the `pc_gens` to a new set of generators of
+        another FreeGroup.
+        This is useful in the implementation of the collection
+        algorithm where a word is converted to a form
+        where each element in the word is a distince generator.
+
+        '''
+        len_pc_gens = len(self.pc_gens)
+        fr_grp_str = ""
+        for i in range(0, 3):
+            fr_grp_str = fr_grp_str + "x_" + str(i) + ", "
+        fr_grp_str = fr_grp_str[:-2]
+
+        # Define a few group on a new set of generators.
+        free_group = free_group(fr_grp_str)
```

Comment:
Or just `free_group(",".join(["x_"+str(i) for i in range(3)])`. Why only 3 generators? And I'm not entirely convinced this method is necessary at all. Why can't we do without?

Reply:
That shouldn't be 3. It should be equal to the length of `pc_gens`. This method would be useful to map the `pc_gens` to the generators of another group. Basically, the `pc_gens` defined on a FreeGroup `F` could be of the form `x**2` or `x*y*x**3`. This could be a problem during the implementation of the collection algorithm as the collection algorithm is implemented on a basis that the pc_gens are of the form `x1`, `x2`...etc(individual entities). So, we define a new `FreeGroup` on generators `x1`, `x2`...etc and map them to the `pc_gens`, in this case, `x1 = x**2` and `x2 = x*y`. 

Now the all the elements in the power presentation and the uncollected subword will be wriiten in the form of new generators. For, example a word `x**5*y` would be written as `x1**2*x2`.